### PR TITLE
feat(lint): add linting rules

### PR DESCRIPTION
**Pull request by annawen1**
### Related Ticket(s)

Add linting to NextJS test application#1937

### Description

Add `eslint`, `stylelint`, and `husky` to our NextJs app

### Changelog

**New**

- add eslint for our `pages` folder and stylelint for our `styles` folder
- add `husky` for our pre-commit hooks

### CODE REVIEW

File: eslint-config-ibmdotcom/index.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
+/**
```
Comment (jeffchew) [inline eslint-config-ibmdotcom/index.js:1]: @annawen1 @kennylam just a thought, maybe not part of this PR. Any thought on publishing our eslint config to npm?

Comment (annawen1) [reply eslint-config-ibmdotcom/index.js:1]: yeah we can do that what about stylelint as well?

Comment (jeffchew) [reply eslint-config-ibmdotcom/index.js:1]: possibly...it would bring consistency for sure, we can also have these as available packages to adopters too if they so choose, though what we are providing is very opinionated

Comment (kennylam) [reply eslint-config-ibmdotcom/index.js:1]: I thought about this. It can't hurt and would save us from duping it in future projects like this. I'm all for it.
